### PR TITLE
Splitted builds into pr and not pr, added ghactions to dependabot and other minors

### PR DESCRIPTION
**Pull request by lfgcampos**
This PR splits our current approach for building into 2.
- master
  - build on master/main/tags
  - will deploy to sonatype
- pullrequest
  - build on PRs only
  - won't deploy to sonatype

Also added a small improvement to dependabot to also check for new versions of our GHActions.
Also updated some outdated action versions.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -50,7 +47,7 @@ jobs:
           clean verify \
           sonar:sonar \
           -Dsonar.projectKey=AxonIQ_axonserver-connector-java \
-          -Dsonar.organization=axoniq \
+          -Dsonar.organization=axonframework \
```
Comment (smcvb) [inline .github/workflows/main.yml:50]: I believe this should still be `axoniq`

File: .github/workflows/pullrequest.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            ${{ runner.os }}-maven
+
+      - name: Maven operation with Sonar
+        if: matrix.sonar-enabled
+        run: |
+          mvn -B -U -Pcoverage \
+          clean verify \
+          sonar:sonar \
+          -Dsonar.projectKey=AxonIQ_axonserver-connector-java \
+          -Dsonar.organization=axonframework \
```
Comment (smcvb) [inline .github/workflows/pullrequest.yml:46]: The `organization` should be `axoniq` here too.

**Review by smcvb — CHANGES_REQUESTED** [review]
I believe the organization parameters for the Sonar invocation should point to `axoniq` i.o. `axonframework`, since this repo belongs to AxonIQ directly.

**Review by smcvb — CHANGES_REQUESTED** [review]
I believe the organization parameters for the Sonar invocation should point to `axoniq` i.o. `axonframework`, since this repo belongs to AxonIQ directly.

File: .github/workflows/main.yml (hunk 1 above)
Comment (lfgcampos) [reply .github/workflows/main.yml]: copy-paste problems =P

File: .github/workflows/pullrequest.yml (hunk 2 above)
Comment (lfgcampos) [reply .github/workflows/pullrequest.yml]: copy-paste problems =P

**Review by smcvb — APPROVED** [review]
Looks good to me :+1:

**Pull request merged**